rasa/nlu/featurizers/dense_featurizer/mitie_featurizer.py

```diff
@@ -1,89 +1,135 @@
+from rasa.shared.nlu.training_data.features import Features
+from rasa.shared.nlu.training_data.message import Message
+from rasa.shared.nlu.constants import FEATURE_TYPE_SENTENCE, FEATURE_TYPE_SEQUENCE
+from rasa.shared.nlu.training_data.training_data import TrainingData
 
 if typing.TYPE_CHECKING:
     import mitie
 
+logger = logging.getLogger(__name__)
 
-class MitieFeaturizer(DenseFeaturizer):
-    @classmethod
-    def required_components(cls) -> List[Type[Component]]:
-        return [MitieNLP, Tokenizer]
+# TODO: This is a workaround around until we have all components migrated to
+# `GraphComponent`.
+MitieFeaturizer = MitieFeaturizer
 
-    defaults = {
-        # Specify what pooling operation should be used to calculate the vector of
-        # the complete utterance. Available options: 'mean' and 'max'
-        POOLING: MEAN_POOLING
-    }
 
-    def __init__(self, component_config: Optional[Dict[Text, Any]] = None) -> None:
-        super().__init__(component_config)
+class MitieFeaturizerGraphComponent(DenseFeaturizer2, GraphComponent):
+    """A class that featurizes using Mitie."""
 
-        self.pooling_operation = self.component_config["pooling"]
+    @staticmethod
+    def get_default_config() -> Dict[Text, Any]:
+        """Returns the component's default config."""
+        return {
+            **DenseFeaturizer2.get_default_config(),
+            # Specify what pooling operation should be used to calculate the vector of
+            # the complete utterance. Available options: 'mean' and 'max'
+            POOLING: MEAN_POOLING,
+        }
 
-    @classmethod
-    def required_packages(cls) -> List[Text]:
+    @staticmethod
+    def required_packages() -> List[Text]:
+        """Any extra python dependencies required for this component to run."""
         return ["mitie", "numpy"]
 
-    def ndim(self, feature_extractor: "mitie.total_word_feature_extractor") -> int:
-        return feature_extractor.num_dimensions
+    def __init__(
+        self, config: Dict[Text, Any], execution_context: ExecutionContext,
+    ) -> None:
+        """Instantiates a new `MitieFeaturizerGraphComponent` instance."""
+        super().__init__(execution_context.node_name, config)
+        self.pooling_operation = self._config[POOLING]
 
-    def train(
-        self,
-        training_data: TrainingData,
-        config: Optional[RasaNLUModelConfig] = None,
-        **kwargs: Any,
+    @classmethod
+    def create(
+        cls,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> "MitieFeaturizerGraphComponent":
+        """Creates a new untrained component (see parent class for full docstring)."""
+        return cls(config, execution_context)
+
+    @classmethod
+    def validate_config(cls, config: Dict[Text, Any]) -> None:
+        """Validates that the component is configured properly."""
+        pass
```

Comment:
Is this + `validate_compatibility_with_tokenizer` implementations to be added later?